api_tests/subscriptions/views/test_subscriptions_detail.py

```diff
@@ -0,0 +1,55 @@
+import pytest
+
+from api.base.settings.defaults import API_BASE
+from osf_tests.factories import AuthUserFactory, NotificationSubscriptionFactory
+
+
+@pytest.mark.django_db
+class TestSubscriptionDetail:
```

Comment:
Add assert for correct status code to each of these

Reply:
Ideally this would cover every combo and can be combined into a single test to speed everything up:

1. GET Valid ID:
 - Invalid user, No user, Valid user
2. GET Invalid ID:
 - Invalid user, No user, Valid user
3. POST Valid ID, Invalid Data
 - Invalid user, No user, Valid user
4. POST Invalid ID
 - Invalid user, No user, Valid user
5. POST Valid ID, Valid Data
 - Invalid user, No user, Valid user **** Note Valid user is last because this mutates the data

(I think that covers everything fully then, can let me know if I missed something)